bolt-sidecar/src/driver.rs

```diff
@@ -60,15 +63,23 @@ impl fmt::Debug for SidecarDriver<StateClient, BlsSigner, PrivateKeySigner> {
     }
 }
 
-impl SidecarDriver<StateClient, BlsSigner, PrivateKeySigner> {
+impl SidecarDriver<StateClient, PrivateKeySigner> {
     /// Create a new sidecar driver with the given [Config] and default components.
     pub async fn new(cfg: Config) -> eyre::Result<Self> {
         // The default state client simply uses the execution API URL to fetch state updates.
         let state_client = StateClient::new(cfg.execution_api_url.clone());
 
-        // Constraints are signed with a BLS private key, for now this is provided
-        // via CLI argument but this is expected to change soon.
-        let constraint_signer = BlsSigner::new(cfg.private_key.clone().unwrap());
+        // Constraints are signed with a BLS private key or Commit-Boost
+        let constraint_signer = if let Some(private_key) = cfg.private_key.clone() {
+            BlsSignerType::PrivateKey(BlsSigner::new(private_key))
+        } else {
+            let commit_boost_client = CommitBoostClient::new(
+                cfg.commit_boost_address.clone().expect("CommitBoost URL must be provided"),
+                &cfg.commit_boost_jwt_hex.clone().expect("CommitBoost JWT must be provided"),
+            )
+            .await?;
+            BlsSignerType::CommitBoost(commit_boost_client)
+        };
```

Comment:
It would be nice to refactor the signing config (`SigningOpts`) to avoid unwrapping CLI options here.
can be done in a separate PR.